lib/logstash/outputs/s3/temporary_file.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: utf-8
+require "thread"
+require "forwardable"
+require "fileutils"
+
+module LogStash
+  module Outputs
+    class S3
+      # Wrap the actual file descriptor into an utility classe
+      # It make it more OOP and easier to reason with the paths.
+      class TemporaryFile
+        extend Forwardable
+        DELEGATES_METHODS = [:path, :write, :close, :size, :fsync]
+
+        def_delegators :@fd, *DELEGATES_METHODS
```

Comment:
AFAICT we only use a constant here to test if these methods are defined in a later spec. I would just define them inline and remove the spec. I don't think there's a ton of value in specs that check that methods merely exist since at that point we're testing that `Forwardable` isn't broken. 

I don't think we need to do any testing here FWIW and we can simplify this.